### PR TITLE
Rewrite optional usage to allow Flowable implementation

### PR DESCRIPTION
**Pull request by nikitin-da**

### CODE REVIEW

File: storio-common/src/main/java/com/pushtorefresh/storio2/operations/internal/FlowableOnSubscribeExecuteAsBlockingOptional.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public FlowableOnSubscribeExecuteAsBlockingOptional(@NonNull PreparedOperation<Result, Optional<Result>, Data> preparedOperation) {
+        this.preparedOperation = preparedOperation;
+    }
+
+    @Override
+    public void subscribe(@NonNull FlowableEmitter<Optional<Result>> emitter) throws Exception {
+        try {
+            final Result value = preparedOperation.executeAsBlocking();
+            emitter.onNext(Optional.of(value));
+            emitter.onComplete();
```
Comment (geralt-encore) [inline storio-common/src/main/java/com/pushtorefresh/storio2/operations/internal/FlowableOnSubscribeExecuteAsBlockingOptional.java]: Are you sure that it has to complete here?

File: storio-content-resolver/src/main/java/com/pushtorefresh/storio2/contentresolver/operations/get/PreparedGetObject.java [hunk 2]
```diff
@@ -56,7 +56,7 @@
     @WorkerThread
     @NonNull
```
Comment (geralt-encore) [inline storio-content-resolver/src/main/java/com/pushtorefresh/storio2/contentresolver/operations/get/PreparedGetObject.java]: @Nullable ?

File: storio-sqlite/src/main/java/com/pushtorefresh/storio2/sqlite/operations/get/PreparedGetObject.java [hunk 3]
```diff
@@ -87,20 +87,20 @@
...
     @NonNull
     @Override
     protected Interceptor getRealCallInterceptor() {
-        return new RealCallInterceptor<T>();
+        return new RealCallInterceptor();
     }
 
-    private class RealCallInterceptor<UnwrappedResult> implements Interceptor {
+    private class RealCallInterceptor implements Interceptor {
         @NonNull
```
Comment (geralt-encore) [inline storio-sqlite/src/main/java/com/pushtorefresh/storio2/sqlite/operations/get/PreparedGetObject.java]: @Nullable

File: storio-common/src/main/java/com/pushtorefresh/storio2/operations/internal/FlowableOnSubscribeExecuteAsBlockingOptional.java (hunk 1 above)
Comment (nikitin-da) [reply storio-common/src/main/java/com/pushtorefresh/storio2/operations/internal/FlowableOnSubscribeExecuteAsBlockingOptional.java]: I think yes. We use this flowable only for starting emission or if have no tables/tags to observe. 
Another flowable created by `ChangesFilter.applyForTablesAndTags` will not complete

**Comment by nikitin-da** [comment]
I will add tests for `RxJavaUtils` tomorrow 
Hope codecov take pity on me then)

**Pull request merged**